test/unit/MatchingEngine.spec.ts

```diff
@@ -248,12 +248,27 @@ describe('MatchingEngine.removePeerOrders', () => {
     const order = createPeerOrder(5, quantity, false);
     engine.addPeerOrder(order);
 
-    let removedOrder = engine.removePeerOrder(order.id, quantityToRemove).order;
+    let removedOrder = engine.removePeerOrder(order.id, order.peerPubKey, quantityToRemove).order;
     expect(removedOrder.quantity).to.be.equal(quantityToRemove);
 
-    removedOrder = engine.removePeerOrder(order.id).order;
+    removedOrder = engine.removePeerOrder(order.id, order.peerPubKey).order;
     expect(removedOrder.quantity).to.be.equal(quantity - quantityToRemove);
   });
+
+  it('should add a new peerOrder and then not allow other peer to remove it', () => {
```

Comment:
Since this test suite doesn't really have knowledge of other peers, maybe just describe this test case as "should not remove a peer order using the wrong peer pub key". Thanks for adding this test case though.